src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFillTests.cs

```diff
@@ -0,0 +1,330 @@
+        }
+
+        [Theory]
+        [InlineData("int", "1")]
+        [InlineData("float", "-0.0f")]
+        [InlineData("double", "-0.0")]
+        [InlineData("decimal", "-0.0m")]
+        [InlineData("string", "\"\"")]
+        [InlineData("int?", "0")]
+        [InlineData("int?", "default(int)")]
+        [InlineData("DateTime?", "new DateTime()")]
+        [InlineData("DateTime?", "default(DateTime)")]
+        [InlineData("DateTimeOffset", "new DateTime()")]
+        [InlineData("DateTimeOffset", "default(DateTime)")]
+        public async Task TestNonDefaultValue(string type, string value)
+        {
+            string source = $@"
+using System;
+
+class C
+{{
+    void M(Span<{type}> span)
+    {{
+        span.Fill({value});
+    }}
+}}
+";
+            await VerifyCSCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task TestGeneric_Unconstrained()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    void M<T>(Span<T> span)
+    {
+        [|span.Fill(default)|];
+    }
+}
+";
+            string expected = @"
+using System;
+
+class C
+{
+    void M<T>(Span<T> span)
+    {
+        span.Clear();
+    }
+}
+";
+            await VerifyCSCodeFixAsync(source, expected);
+        }
+
+        [Fact]
+        public async Task TestGeneric_Reference_Null()
+        {
+            string source = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : class
+    {
+        [|span.Fill(null)|];
+    }
+}
+";
+            string expected = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : class
+    {
+        span.Clear();
+    }
+}
+";
+            await VerifyCSCodeFixAsync(source, expected);
+        }
+
+        [Fact]
+        public async Task TestGeneric_ValueType_Null()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : struct
+    {
+        [|span.Fill(null)|];
+    }
+}
+";
+            string expected = @"
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : struct
+    {
+        span.Clear();
+    }
+}
+";
+            await VerifyCSCodeFixAsync(source, expected);
+        }
+
+        [Fact]
+        public async Task TestGeneric_Reference_DefaultT()
+        {
+            string source = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : class
+    {
+        [|span.Fill(default(T))|];
+    }
+}
+";
+            string expected = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : class
+    {
+        span.Clear();
+    }
+}
+";
+            await VerifyCSCodeFixAsync(source, expected);
+        }
+
+        [Fact]
+        public async Task TestGeneric_ValueType_DefaultT()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : struct
+    {
+        span.Fill(default(T));
```

Comment:
Why it is not flagging this?

Reply:
Because `default(T)` is not `default(Nullable<T>)` which is `null`. Notice the struct constraint.